addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -269,8 +297,18 @@ def get_vals(member):
                     bus_last_id = self.env["bus.bus"].sudo()._bus_last_id()
                     diff.append({"message_unread_counter_bus_id": bus_last_id})
                 store.add(member, diff).bus_send()
+            if diffToChannel:
+                store = Store(bus_channel=member.channel_id)
+                store.add(member, diffToChannel).bus_send()
         return result
 
+    @api.model
+    def _sync_field_names_within_channel(self):
+        return [
+            # sudo: discuss.channel.member - reading channel ownership related to a member is considered acceptable
+            Store.Attr("channel_role", sudo=True),
```

Comment:
sync does not seem to happen as expected when I use the set/unset from the avatar card